### ISSUE TITLE
Parallel experiments will try to delete the same test file on remote storage

### ISSUE THREAD

**Issue by mads-oestergaard** [allegroai/clearml#897]
## Describe the bug
The storage helper will try to verify that we have sufficient rights by writing a dummy file and deleting it again. However, the name of the file is hardcoded to `[...]/.clearml.test`, see [this line](https://github.com/allegroai/clearml/blob/b3e330ea9b00e081988c2effe3a3c62e3824e0a3/clearml/storage/helper.py#L993). 

If you run multiple experiments in parallel locally using e.g. hydras multirun and joblib, then only the first process that manages to delete the file will continue to run, the others will fail. 

A simple fix could be to append some random identifier to the test file, to make sure that multiple parallel experiments do not try to delete the same file at the same time. 

## To reproduce
python file:
```python
import hydra
from omegaconf import DictConfig
from clearml import Task
from clearml.storage.helper import StorageHelper
from clearml.backend_config.bucket_config import AzureContainerConfig

def get_storage_uri() -> str:
    # Set up logging to Azure
    bucket_config = AzureContainerConfig(
        account_name="💩",  # type: ignore
        account_key="💩",  # type: ignore
        container_name="💩",  # type: ignore
    )
    StorageHelper.add_azure_configuration(bucket_config)
    return StorageHelper.get_azure_storage_uri_from_config(bucket_config)


def init_experiment_tracker(config: DictConfig) -> Task:
    """Initialize the experiment tracker and return the task and
    (possibly modified) configuration object."""

    # Initialize the task
    storage_uri = get_storage_uri()
    task: Task = Task.init(
        project_name=config.clearml.project_name,
        task_name=config.clearml.task_name,
        task_type=Task.TaskTypes.training,
        reuse_last_task_id=False,
        output_uri=storage_uri,
    )
    task.logger.set_default_upload_destination(storage_uri)
    return task


@hydra.main(version_base=None, config_path="conf", config_name="config")
def main(config: DictConfig) -> None:
    init_experiment_tracker(config)

    # this guy is what we'll change
    print(config.sweep_param)


if __name__ == "__main__":
    main()
```

and in conf/config.yaml next to the python file: 
```yaml
defaults:
  - override hydra/launcher: joblib
  - _self_

clearml:
  project_name: bugs
  task_name: example

sweep_param: 0

hydra:
  launcher:
    n_jobs: 2
  mode: MULTIRUN
  sweeper:
    params:
      sweep_param: 0, 1, 2, 3
```

This requires:
- hydra-core == 1.2.0
- hydra-joblib-launcher == 1.2.0
- clearml == 1.7.0

## Expected behaviour
All parallel experiments should be writing their own files to test the storage connection to avoid race conditions. Sometimes however, timing makes the folllowing exception: 

```
...
Error executing job with overrides: ['sweep_param=1']
Traceback (most recent call last):
  File "/home/aia0/Repos/💩/.venv/lib/python3.9/site-packages/hydra/_internal/utils.py", line 213, in run_and_report
    return func()
  File "/home/aia0/Repos/💩/.venv/lib/python3.9/site-packages/hydra/_internal/utils.py", line 461, in <lambda>
    lambda: hydra.multirun(
  File "/home/aia0/Repos/💩/.venv/lib/python3.9/site-packages/hydra/_internal/hydra.py", line 162, in multirun
    ret = sweeper.sweep(arguments=task_overrides)
  File "/home/aia0/Repos/💩/.venv/lib/python3.9/site-packages/hydra/_internal/core_plugins/basic_sweeper.py", line 182, in sweep
    _ = r.return_value
  File "/home/aia0/Repos/💩/.venv/lib/python3.9/site-packages/hydra/core/utils.py", line 260, in return_value
    raise self._return_value
ValueError: Insufficient permissions (delete failed) for azure://💩.blob.core.windows.net/💩
```

## Environment
* Server type: self hosted
* ClearML SDK Version: 1.7.0
* ClearML Server Version: 1.7.0-232 
* Python Version: 3.9.5
* OS: Linux 

## Related Discussion
.

**Comment by ainoam** [comment]
Appreciate the report @mads-oestergaard!
Indeed such race conditions should be accommodated. Ideally, the access privilege test would be performed only once and not for every experiment instance.

**Comment by pollfly** [comment]
Hey @mads-oestergaard! Just letting you know that this issue has been resolved in the recently released [v1.11.0](https://github.com/allegroai/clearml/releases/tag/v1.11.0). Let us know if there are any issues :)

**Comment by jkhenning** [comment]
@mads-oestergaard closing this. Please reopen if required.
